prediction_market_agent/agents/microchain_agent/app.py

```diff
@@ -0,0 +1,108 @@
+# Imports using asyncio (in this case mech_client) cause issues with Streamlit
+from prediction_market_agent.streamlit_utils import (  # isort:skip
+    streamlit_asyncio_event_loop_hack,
+)
+
+streamlit_asyncio_event_loop_hack()
+
+import streamlit as st
+from microchain import Agent
+from prediction_market_agent_tooling.markets.markets import MarketType
+from prediction_market_agent_tooling.tools.costs import openai_costs
+from prediction_market_agent_tooling.tools.utils import check_not_none
+
+from prediction_market_agent.agents.microchain_agent.microchain_agent import get_agent
+from prediction_market_agent.agents.microchain_agent.utils import (
+    has_been_run_past_initialization,
+)
+from prediction_market_agent.streamlit_utils import check_required_api_keys
+from prediction_market_agent.utils import APIKeys
+
+
+def check_api_keys() -> None:
+    keys = APIKeys()
+    if not keys.OPENAI_API_KEY:
+        st.error("No OpenAI API Key provided via env var/secret.")
+        st.stop()
+    elif not keys.BET_FROM_PRIVATE_KEY:
+        st.error("No wallet private key provided via env var/secret.")
+        st.stop()
+
+
+def run_agent(agent: Agent, iterations: int, model: str) -> None:
+    with openai_costs(model) as costs:
+        with st.spinner("Agent is running..."):
+            for _ in range(iterations):
+                agent.run(iterations=1, resume=True)
```

Comment:
When calling agent.run(iterations=1), I'm worried the history (agent.history) is overwritten, i.e. it has no idea of the previous actions it took earlier.
So you might want to track the agent.history via an external variable, e.g. 
```
temp.history = []
for _ in range(iterations):
    agent.run(iterations=1) # ToDo - Add history to agent (note that it will be added to the prompt)
    temp_history = agent.history
```

Reply:
The `resume=True` flag means that history isn't overwritten (see this microchain PR here https://github.com/galatolofederico/microchain/pull/10)

Reply:
Excellent, thanks